content/departments/cloud/technical-docs/private-code-hosts/index.md

```diff
@@ -0,0 +1,153 @@
+
+This solution is recommended by Cloud Team, but only possible if customers' code hosts can be accessible publicly and customer is able to allow incoming traffic from Sourcegraph-owned static IP addresses.
+Outgoing traffic of Cloud instances goes through Cloud NAT with stable IPs. All IPs are reservered exclusively on a per customer basis.
+[More informations about IP allowlist](../../index.md#faq-what-is-the-cloud-instance-ip)
+
+If customer is not able to meet these requirements, Sourcegraph can propose [VPN solution for code hosts deployed on AWS](#aws-gcp-site-to-site-vpn-solution)
+
+## AWS GCP site to site VPN solution (AWS code hosts only)
+
+### Architecture
+
+[AWS VPN architecture](https://app.excalidraw.com/s/4Dr1S6qmmY7/4L5TAaxiYAy)
+
+### Creation process
+
+AWS GCP VPN extension is connecting existing Cloud Instance with customer dedicated AWS account (maintaned by Cloud Team).
+
+#### 1. Modify `config.yaml` with additional section:
+
+```yaml
+spec:
+  infrastructure:
+    privateCodeHost:
+      aws:
+        accountID: <AWS_ACCOUNT_ID>
+        highAvailable: true
+        region: <AWS_REGION>
+      enabled: true
+      type: awsvpn
+```
+
+#### 2. Generate additional terraform stacks
+
+```
+mi2 generate cdktf
+```
+
+this will generate `awsvpc` and `awsvpn` stacks.
+
+Creation process detail walkthrough:
+
+1. AWS VPC is created in customer dedicated Sourcegraph AWS Account.
+1. Cloud Router with GCP-side BGP ASN number is created in Managed Instance VPC.
+1. AWS VPN Gateway with BGP AWS-side ASN number is created. It is attached to VPC and uses VPC route tables to propagate BPG sessions.
+1. High Available GCP VPN Gateway is created in Managed Instance GCP VPC with 2 interfaces.
+1. AWS Customer Gateway is created (uses GCP-side BGP ASN). For high available options 2 gateways are created.
+1. AWS VPN connection is created with 2 tunnel options (every tunnel uses different 32B PreSharedKey). For high available options 2 VPC connections are created.
+1. GCP External VPN Gateway is created with interfaces correspoding for each AWS VPN connection tunnel. For high available options External VPN Gateway has 4 interfaces, for non high-available 2.
+1. For each External VPN Gateway interface GCP VPN Tunnel is created. It uses dedicated PreSharedKey (same in on AWS VPN Connection(s)).
+1. For each GCP VPN Tunnel interface is added to Cloud Router. It uses IP range correspoding to AWS VPN Connection tunnel internal address.
+1. For each AWS VPN Connection tunnel internal address BGP peer is added to Cloud Router.
+
+For more details, go to [Google documentation](https://cloud.google.com/network-connectivity/docs/vpn/tutorials/create-ha-vpn-connections-google-cloud-aws)
+
+### VPN Verification
+
+For each customer using private code host, additional section to dashboard.md is added.
+Can be generated on deman via:
+
+```sh
+mi2 instance dashboard --output <FILE_NAME>.md
+```
+
+Secction example:
```

Comment:
I think we can remove everything below and just say, see the dashboard to avoid maintaining two source of true

Reply:
Make sense, we would need to duplicate it anyway 👍